elastic/datadog_checks/elastic/elastic.py

```diff
@@ -439,6 +461,123 @@ def _process_cat_allocation_data(self, admin_forwarder, version, base_tags):
                 desc = CAT_ALLOCATION_METRICS[metric]
                 self._process_metric(cat_allocation_dic, metric, *desc, tags=tags)
 
+    def _process_custom_metric(
+        self, value, data_path, value_path, dynamic_tags, xtype, metric_name, tags=None, hostname=None
+    ):
+        """
+        value: JSON payload to traverse
+        data_path: path to data right before metric value or right before list of metric values
+        value_path: path to data after data_path to metric value
+        dynamic_tags: list of tags and their value_paths
+        xtype: datadog type
+        metric_name: datadog metric name
+        tags: a lambda to apply to the numerical value
+        hostname: a lambda to apply to the numerical value
```

Comment:
Why is hostname needed?

Reply:
I originally included it to be consistent with the `_process_metric()` function, although since I'm currently not using it I can remove it for `_process_custom_metric()`.